CMakeLists.txt

```diff
@@ -1,39 +1,61 @@
 cmake_minimum_required(VERSION 3.15 FATAL_ERROR)
+
+include(gen/CurrentBuildVulkanVersion.cmake)
+
 project(VulkanBootstrap
     LANGUAGES CXX
-    DESCRIPTION "A Vulkan utility library to ease the initialization steps in Vulkan")
+    DESCRIPTION "A Vulkan utility library to ease the initialization steps in Vulkan"
+    VERSION ${VK_BOOTSTRAP_SOURCE_HEADER_VERSION})
 
 option(VK_BOOTSTRAP_DISABLE_WARNINGS "Disable warnings during compilation" OFF)
 option(VK_BOOTSTRAP_WERROR "Enable warnings as errors during compilation" OFF)
 
-add_library(vk-bootstrap-vulkan-headers INTERFACE)
+if (CMAKE_VERSION VERSION_LESS "3.21")
+    # By default, only build tests & install if this repo is standalone, but still allow configuration by the user
+    if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+        set(PROJECT_IS_TOP_LEVEL ON)
+    else()
+        set(PROJECT_IS_TOP_LEVEL OFF)
+    endif()
+endif()
+
+option(VK_BOOTSTRAP_TEST "Test Vk-Bootstrap using Catch2 as well as build examples" ${PROJECT_IS_TOP_LEVEL})
+option(VK_BOOTSTRAP_INSTALL "Enable warnings as errors during compilation" ${PROJECT_IS_TOP_LEVEL})
 
 set(VK_BOOTSTRAP_VULKAN_HEADER_DIR "" CACHE STRING "Specify the location of the Vulkan-Headers include directory.")
```

Comment:
Instead of having the type by a STRING it can be a directory. This matters for cmake-gui